src/main/java/cceclipseplugin/core/EclipseRequestManager.java

```diff
@@ -0,0 +1,201 @@
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.swt.widgets.Display;
+
+import cceclipseplugin.ui.UIRequestErrorHandler;
+import cceclipseplugin.ui.dialogs.DialogStrings;
+import cceclipseplugin.ui.dialogs.MessageDialog;
+import dataMgmt.DataManager;
+import dataMgmt.models.FileMetadata;
+import dataMgmt.models.ProjectMetadata;
+import requestMgmt.IInvalidResponseHandler;
+import requestMgmt.RequestManager;
+import websocket.IRequestSendErrorHandler;
+import websocket.WSManager;
+import websocket.models.File;
+import websocket.models.Project;
+import websocket.models.Request;
+import websocket.models.requests.FileCreateRequest;
+import websocket.models.requests.FilePullRequest;
+import websocket.models.responses.FileCreateResponse;
+import websocket.models.responses.FilePullResponse;
+
+public class EclipseRequestManager extends RequestManager {
+
+	public EclipseRequestManager(DataManager dataManager, WSManager wsManager,
+			IRequestSendErrorHandler requestSendErrorHandler, IInvalidResponseHandler invalidResponseHandler) {
+		super(dataManager, wsManager, requestSendErrorHandler, invalidResponseHandler);
+	}
+	
+	@Override
+	public void finishSubscribeToProject(long id, File[] files) {
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+		Project p = PluginManager.getInstance().getDataManager().getSessionStorage().getProjectById(id);
+		IProject eclipseProject = root.getProject(p.getName());
+		ProjectMetadata meta =  new ProjectMetadata();
+		meta.setName(p.getName());
+		meta.setProjectID(id);
+		PluginManager.getInstance().getMetadataManager().putProjectMetadata(eclipseProject.getFullPath().toString(), meta);
+		try {
+			if (eclipseProject.exists()) {
+				eclipseProject.delete(true, true, new NullProgressMonitor());
+			}
+			eclipseProject.create(new NullProgressMonitor());
+			eclipseProject.open(new NullProgressMonitor());
+			for (File f : files) {
+				pullFileAndCreate(eclipseProject, p, f, new NullProgressMonitor());
+			}
+		} catch (CoreException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	private byte[] fileBytes;
+	
+	public void pullFileAndCreate(IProject p, Project ccp, File file, IProgressMonitor progressMonitor) {
+		Request req = (new FilePullRequest(file.getFileID())).getRequest(response -> {
+				if (response.getStatus() == 200) {
+					fileBytes = ((FilePullResponse) response.getData()).getFileBytes();
+					
+					String path = file.getRelativePath();
+					Path relPath = new Path(path);
+					if (!path.equals("") && !path.equals(".")) {
+						String currentFolder = "";
+						for (int i = 0; i < relPath.segmentCount(); i++) {
+							// iterate through path segments and create if they don't exist
+							currentFolder += "/" + relPath.segment(i);
```

Comment:
Paths.get doesn't create the folders on disk if they don't exist does it? That's the main reason why I'm iterating through it. Also the Path class used here in this code isn't the Java implementation of Path, it's the Eclipse one. If there's something I'm missing here let me know.